src/checks_windows.go

```diff
@@ -204,10 +216,50 @@ func packageInstalled(packageName string) (bool, error) {
 }
 
 func serviceUp(serviceName string) (bool, error) {
-	return commandOutput("(Get-Service -Name '"+serviceName+"').Status", "Running")
+	serviceStatus, err := getLocalServiceStatus(serviceName)
+	return serviceStatus.IsRunning, err
+}
+
+func serviceStatus(serviceName, wantedStatus, startupType string) (bool, error) {
+	status, err := getLocalServiceStatus(serviceName)
+	var boolWantedStatus bool
+	if err != nil {
+		return false, err
+	}
+	wantedStatus = strings.ToLower(wantedStatus)
+	if wantedStatus == "running" {
+		boolWantedStatus = true
+	} else if wantedStatus == "stopped" {
+		boolWantedStatus = false
+	} else {
+		return false, nil
+	}
+	if status.IsRunning == boolWantedStatus {
+		serviceKey := `HKLM\SYSTEM\CurrentControlSet\Services\` + serviceName
+		var wantedStartupTypeNumber string
+		startupType = strings.ToLower(startupType)
+		if startupType == "automatic" {
+			wantedStartupTypeNumber = "2"
+		} else if startupType == "manual" {
+			wantedStartupTypeNumber = "3"
+		} else if startupType == "disabled" {
+			wantedStartupTypeNumber = "4"
+		} else {
+			return false, nil
```

Comment:
Should return an error here for unknown startUp type (with a failPrint)

Reply:
safinlint: unhandled error: try: ```go
failPrint("Invalid startup type " + startupType)
return false, errors.New("Invalid startup type " + startupType)
```

Reply:
switch case?